Sources/Yams/Representer.swift

```diff
@@ -160,16 +199,82 @@ extension Optional: NodeRepresentable {
     }
 }
 
-extension Array: NodeRepresentable {
+extension Decimal: ScalarRepresentable {
     public func represented() throws -> Node {
-        let nodes = try map(represent)
-        return Node(nodes, Tag(.seq))
+        return Node(description)
     }
 }
 
-extension Dictionary: NodeRepresentable {
+extension URL: ScalarRepresentable {
     public func represented() throws -> Node {
-        let pairs = try map { (key: try represent($0.0), value: try represent($0.1)) }
-        return Node(pairs.sorted { $0.key < $1.key }, Tag(.map))
+        return Node(absoluteString)
     }
 }
+
+#if swift(>=4.0)
+
+/// MARK: - ScalarRepresentableCustomizedForCodable
+
+public protocol ScalarRepresentableCustomizedForCodable: ScalarRepresentable {
+    func representedForCodable() -> Node
+}
+
+extension Date: ScalarRepresentableCustomizedForCodable {
+    public func representedForCodable() -> Node {
+        return Node(iso8601StringWithFullNanosecond, Tag(.timestamp))
+    }
+}
+
+extension Double: ScalarRepresentableCustomizedForCodable {
```

Comment:
You should be able to combine these two implementations:

```diff
diff --git a/Sources/Yams/Representer.swift b/Sources/Yams/Representer.swift
index 09e05f9..c79eb44 100644
--- a/Sources/Yams/Representer.swift
+++ b/Sources/Yams/Representer.swift
@@ -225,23 +225,25 @@ extension Date: ScalarRepresentableCustomizedForCodable {
     }
 }
 
-extension Double: ScalarRepresentableCustomizedForCodable {
+extension Double: ScalarRepresentableCustomizedForCodable {}
+extension Float: ScalarRepresentableCustomizedForCodable {}
+
+extension FloatingPoint where Self: CVarArg {
     public func representedForCodable() -> Node {
-    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
-        return Node(formattedStringForCodable, Tag(.float))
-    #else
+#if os(Linux)
         return Node(doubleFormatter.string(for: self)!.replacingOccurrences(of: "+-", with: "-"), Tag(.float))
-    #endif
+#else
+        return Node(formattedStringForCodable, Tag(.float))
+#endif
     }
 
     private var formattedStringForCodable: String {
         // Since `NumberFormatter` creates a string with insufficient precision for Decode,
         // it uses with `String(format:...)`
-    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+#if os(Linux)
+        let DBL_DECIMAL_DIG = 9
+#endif
         let string = String(format: "%.*g", DBL_DECIMAL_DIG, self)
-    #else
-        let string = String(format: "%.*g", 9, self)
-    #endif
         // "%*.g" does not use scientific notation if the exponent is less than –4.
         // So fallback to using `NumberFormatter` if string does not uses scientific notation.
         guard string.lazy.suffix(5).contains("e") else {
@@ -251,30 +253,4 @@ extension Double: ScalarRepresentableCustomizedForCodable {
     }
 }
 
-extension Float: ScalarRepresentableCustomizedForCodable {
-    public func representedForCodable() -> Node {
-    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
-        return Node(formattedStringForCodable, Tag(.float))
-    #else
-        return Node(floatFormatter.string(for: self)!.replacingOccurrences(of: "+-", with: "-"), Tag(.float))
-    #endif
-    }
-
-    private var formattedStringForCodable: String {
-        // Since `NumberFormatter` creates a string with insufficient precision for Decode,
-        // it uses with `String(format:...)`
-    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
-        let string = String(format: "%.*g", FLT_DECIMAL_DIG, self)
-    #else
-        let string = String(format: "%.*g", 9, self)
-    #endif
-        // "%*.g" does not use scientific notation if the exponent is less than –4.
-        // So fallback to using `NumberFormatter` if string does not uses scientific notation.
-        guard string.lazy.suffix(6).contains("e") else {
-            return floatFormatter.string(for: self)!.replacingOccurrences(of: "+-", with: "-")
-        }
-        return string
-    }
-}
-
 #endif
```

Reply:
Thanks for patch.
By applying this, I noticed that my commits has mistake:
```diff
+#if os(Linux)
+        let DBL_DECIMAL_DIG = 9
+#endif
```
`DBL_DECIMAL_DIG` should be `17`.
That caused errors that fixed on 587c382.